cluster/engine.go

```diff
@@ -0,0 +1,508 @@
+package cluster
```

Comment:
done, I'm not sure why it didn't work with for the test, but at least it worked for the code